.github/workflows/build.yml

```
@@ -0,0 +1,48 @@
+name: build.yml
+
+on:
+  push:
+    branches: [master, branch-*]
+  pull_request:
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: ${{ github.ref_name != github.event.repository.default_branch }}
+
+jobs:
+  build:
+    runs-on: github-ubuntu-latest-s
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - name: Checkout
+        uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
+        with:
+          ref: ${{ github.head_ref || github.ref_name }}
```

```suggestion
        uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
```
I think we can rely on the default behavior here, no?

---

`this defaults to the reference or SHA for that event` -> it checks out the commit hash,
 otherwise uses default branch -> `master`